### ISSUE TITLE
About the optimizer

### ISSUE THREAD

**Issue by chamwen** [yfzhang114/Environment-Label-Smoothing#1]
Hi~
My question is as follows:
In /tree/master/Domain_Adaptation/cdan_sdat.py, the domain_discriminator and classifier used different optimizers. 
However, in cdan.py, they used the same.
Why?

**Comment by yfzhang114** [comment]
The implementation of baselines just follows the paper "A Closer Look at Smoothness in Domain Adversarial Training". CDAN uses GRL (gradient reversal layer) and does not need a different optimizer.

**Comment by chamwen** [comment]
Thanks for your reply! So the cdan_sdat method needs different optimizers? Will this difference affect its performance?

**Comment by yfzhang114** [comment]
It depends on the choice of hyperparameters. Generally speaking, GRL and using two different optimizers will not bring huge performance differences, so they can be substituted for each other at this time. But just like the results obtained from our theoretical analysis in the main text, Appendix A.10.3: GRL generally cannot guarantee non-asymptotic convergence but may have sub-linear convergence capabilities using different optimizers (Appendix A.10.4).
